platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DynamicCallProblemReporting.kt

```diff
@@ -66,17 +68,38 @@ class DefaultDynamicCallProblemReporting : DynamicCallProblemReporting {
     }
 
     override fun leaveDynamicCall(entryPoint: Any) {
+        if (currentThreadState.callStack.isEmpty()) {
+            return
+        }
         val innermostCall = currentThreadState.callStack.pop()
         check(entryPoint == innermostCall.entryPoint) { "Mismatched enter-leave calls in DynamicCallProjectIsolationProblemReporting" }
     }
 
     override fun unreportedProblemInCurrentCall(problemKey: Any): Boolean {
         val currentThreadCallStack = currentThreadState.callStack
-        check(currentThreadCallStack.isNotEmpty()) { "Expected unreportedProblemInCurrentCall to be called after enterDynamicCall" }
+        if (currentThreadCallStack.isEmpty()) {
+            logMissingScope()
+            return true
+        }
+
         return currentThreadCallStack.peek().problemsReportedInCurrentCall.add(problemKey)
     }
 
+    private
+    fun logMissingScope() {
+        if (logger.isEnabled(LogLevel.DEBUG)) {
```

Comment:
```suggestion
        logger.debug {
```

See https://github.com/gradle/gradle/blob/783e4aa305d675d10e2f5f56f2c5794d15356689/platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/Logger.kt#L35